src/query/plan/rule_based_planner.hpp

```diff
@@ -33,8 +33,11 @@ struct PatternComprehensionData {
   PatternComprehensionData(std::shared_ptr<LogicalOperator> lop, Symbol res_symbol)
       : op(std::move(lop)), result_symbol(res_symbol) {}
 
+  PatternComprehensionData(PatternComprehension *pc, Symbol res_symbol)
+      : pattern_comprehension(pc), result_symbol(res_symbol) {}
+
   PatternComprehension *pattern_comprehension = nullptr;
-  std::shared_ptr<LogicalOperator> op;
+  std::shared_ptr<LogicalOperator> op = nullptr;
```

Comment:
You don't need to define it as `nullptr` here, it should be empty: https://en.cppreference.com/w/cpp/memory/shared_ptr/shared_ptr 
Check (1)

Reply:
Thank you, Fico! Will fix this :)